spec/models/miq_group_spec.rb

```diff
@@ -96,63 +96,40 @@
       @miq_group.description = "      leading and trailing white spaces     "
       @miq_group.description.should == "leading and trailing white spaces"
     end
+  end
 
-    it "should return LDAP groups by user name" do
-      auth_config = { :group_memberships_max_depth => 1 }
-      stub_server_configuration(:authentication =>  auth_config)
-
-      miq_ldap = double('miq_ldap')
-      miq_ldap.stub(:fqusername => 'fred')
-      miq_ldap.stub(:normalize => 'fred flintstone')
-      miq_ldap.stub(:bind => true)
-      miq_ldap.stub(:get_user_object => 'user object')
-      memberships = [ 'foo', 'bar' ]
-      miq_ldap.stub(:get_memberships => memberships)
+  describe "#get_ldap_groups_by_user" do
+    before do
+      stub_server_configuration(:authentication => {:group_memberships_max_depth => 1})
+
+      miq_ldap = double('miq_ldap',
+                        :fqusername      => 'fred',
+                        :normalize       => 'fred flintstone',
+                        :bind            => true,
+                        :get_user_object => 'user object',
+                        :get_memberships => %w(foo bar))
       MiqLdap.stub(:new).and_return(miq_ldap)
+    end
 
-      MiqGroup.get_ldap_groups_by_user('user', 'bind_dn', 'password').should == memberships
+    it "should return LDAP groups by user name" do
+      MiqGroup.get_ldap_groups_by_user('fred', 'bind_dn', 'password').should == %w(foo bar)
     end
 
     it "should issue an error message when user name could not be bound to LDAP" do
-      auth_config = { :group_memberships_max_depth => 1 }
-      stub_server_configuration(:authentication =>  auth_config)
-
-      miq_ldap = double('miq_ldap')
-      miq_ldap.stub(:fqusername => 'fred')
-      miq_ldap.stub(:normalize => 'fred flintstone')
-      miq_ldap.stub(:bind => false)
-      miq_ldap.stub(:get_user_object => 'user object')
-      memberships = [ 'foo', 'bar' ]
-      miq_ldap.stub(:get_memberships => memberships)
-      MiqLdap.stub(:new).and_return(miq_ldap)
-
-      lambda {
-              MiqGroup.get_ldap_groups_by_user('user', 'bind_dn', 'password')
-             }.should
-             raise_error( MiqException::MiqEVMLoginError,
-                          "Bind failed for user bind_dn"
-                        )
+      MiqLdap.new.stub(:bind => false)
+      # darn, wanted a MiqException::MiqEVMLoginError
+      expect {
+        MiqGroup.get_ldap_groups_by_user('fred', 'bind_dn', 'password')
+      }.to raise_error(RuntimeError, "Bind failed for user bind_dn")
```

Comment:
History for the `get_ldap_groups_by_user` method in miq_group.rb shows that the `raise` calls have been using `RuntimeError` since 2009.  The tests were added in 2011 and my only guess is that the `raise_error` check has always been ignored.